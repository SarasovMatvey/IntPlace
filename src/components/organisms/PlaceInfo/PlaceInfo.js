import { Header, Image, List } from "semantic-ui-react";
import { Swiper, SwiperSlide } from "swiper/react";
import "./PlaceInfo.sass";
import "swiper/swiper.scss";
import _ from "lodash";

function PlaceInfo({ mainImg, placeName, description, media = [], params }) {
  return (
    <div className="place-info">
      <div className="place-info__base-info">
        <Header as="h1">{placeName}</Header>
        <p className="place-info__desc">{description}</p>
        <List className="place-info__params">
          {_.map(params, (value, key) =>
            value ? (
              <List.Item>
                <List.Header>{key}</List.Header>
                {value}
              </List.Item>
            ) : null
          )}
        </List>
        <Swiper
          className="place-info__swiper"
          spaceBetween={50}
          slidesPerView={media.length >= 2 ? 2 : 1}
          observer
          observeParents
          observeSlideChildren
        >
          {media.map(url => (
            <SwiperSlide>
              {getUrlExtension(url) === "jpg" ? (
                <Image src={url} />
              ) : (
                <video className="place-info__video" controls="controls">
                  <source src={url} />
                </video>
              )}
            </SwiperSlide>
          ))}
        </Swiper>
      </div>
      <Image className="place-info__main-img" src={mainImg} size="big" />
    </div>
  );

  function getUrlExtension(url) {
    return url.split(/[#?]/)[0].split(".").pop().trim();
  }
}

export default PlaceInfo;
