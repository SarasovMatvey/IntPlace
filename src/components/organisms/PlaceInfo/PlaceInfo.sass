.place-info 
  position: relative
  display: grid
  grid-template-columns: repeat(2, 50%)
  &__fav-btn
    position: absolute
    top: .1rem
    left: -1rem
    font-size: 1.5em
    background: none
    border: none
    color: #ff9800
    cursor: pointer
  &__base-info
    padding: 0 2rem
    display: flex
    flex-direction: column
  &__desc
    text-align: justify
  &__main-img 
    height: 100% !important
    object-fit: cover
  &__swiper
    margin-top: auto
    max-height: 300px
  .swiper-wrapper
    display: flex
    align-items: flex-end
    margin-top: 2rem
  &__video
    width: 100%