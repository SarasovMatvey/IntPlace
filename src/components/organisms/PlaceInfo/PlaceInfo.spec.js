import PlaceInfo from "./PlaceInfo";

const setUp = props => shallow(<PlaceInfo {...props} />);

describe("PlaceInfo", () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it("should renders properly", () => {
    expect(component).toMatchSnapshot();
  });
});
