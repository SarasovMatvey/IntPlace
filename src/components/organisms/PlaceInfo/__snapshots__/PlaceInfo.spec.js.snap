// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceInfo should renders properly 1`] = `
<div
  className="place-info "
>
  <button
    className="place-info__fav-btn"
  >
    <Icon
      as="i"
      name="star outline"
    />
  </button>
  <div
    className="place-info__base-info"
  >
    <Header
      as="h1"
    />
    <p
      className="place-info__desc"
    />
    <List
      className="place-info__params"
    />
    <div>
      <Swiper
        className="place-info__swiper"
        observeParents={true}
        observeSlideChildren={true}
        observer={true}
        slidesPerView={1}
        spaceBetween={50}
      />
    </div>
  </div>
  <Image
    as="img"
    className="place-info__main-img"
    size="big"
    ui={true}
  />
</div>
`;
